src/common/src/array/data_chunk.rs

```diff
@@ -579,6 +588,11 @@ impl DataChunkTestExt for DataChunk {
                         return None;
                     }
                     Some(match builder {
+                        ArrayBuilderImpl::Bool(_) => ScalarImpl::Bool(match s {
```

Comment:
Can we remove the default branch of this `match`?

Reply:
You inspired me that we could reuse `ScalarImpl::from_text` for this (except for `f`/`t` ). 😇